### ISSUE TITLE
Request tile editing on protected area to give item back

### ISSUE THREAD

**Issue by tchomphoochan** [Pryaxis/TShock#825]
Yes, as usually, placing anything in protected area will result in tile disappeared, destroyed, consumed. That might be a problem if server is SSC and tile placed is precious one. This also includes editable tile, such as Weapon Rack. Many of players accidently right click on Weapon Rack so their items are consumed. So I request here, if possible and someone will do it, please make it that it gives player back the item, at least just Weapon Rack.

**TL;DR : Give item back on placing tile in protected area**

Optional : For SSC enabled server, there should be a way to prevent noob players who don't know they can't place Piggy Bank/Safe to buy it and place it then just complain later. Maybe on place = money give back? Probably... Or an inventory check if piggy bank has come inside.
